src/core/mesh/qgsmeshplotter.h

```diff
@@ -0,0 +1,213 @@
+    void graduatedColorsDiscrete( double value1, double value2, QList<double> &breakValues, QList<QColor> &breakColors, QList<QLinearGradient> &gradients ) const;
+};
+
+/**
+ * \ingroup core
+ *
+ * Class defining stroke width to render dataset. The width can vary depending on dataset value.
+ *
+ * \since QGIS 3.14
+ */
+class CORE_EXPORT QgsMeshStrokeWidth
+{
+  public:
+    //! Returns the minimum value used to defined the varying stroke width
+    double minimumValue() const;
+    //! Sets the minimum value used to defined the varying stroke width
+    void setMinimumValue( double minimumValue );
+
+    //! Returns the maximum value used to defined the varying stroke width
+    double maximumValue() const;
+    //! Sets the maximum value used to defined the varying stroke width
+    void setMaximumValue( double maximumValue );
+
+    //! Returns the minimum width used to defined the varying stroke width
+    double minimumWidth() const;
+    //! Sets the minimum width used to defined the varying stroke width
+    void setMinimumWidth( double minimumWidth );
+
+    //! Returns the maximum width used to defined the varying stroke width
+    double maximumWidth() const;
+    //! Sets the maximum width used to defined the varying stroke width
+    void setMaximumWidth( double maximumWidth );
+
+    //! Returns whether the varying stroke width ignores out of range value
+    bool ignoreOutOfRange() const;
+    //! Sets whether the varying stroke width ignores out of range value
+    void setIgnoreOutOfRange( bool ignoreOutOfRange );
+
+    //! Returns whether the stroke width is varying
+    bool isWidthVarying() const;
+    //! Returns whether the stroke width is varying
+    void setIsWidthVarying( bool isWidthVarying );
+
+    //! Returns the fixed stroke width
+    double fixedStrokeWidth() const;
+    //! Sets the fixed stroke width
+    void setFixedStrokeWidth( double fixedWidth );
+
+    //! Returns the varying stroke width depending on value, if not varying returns the fixed stroke width
+    double strokeWidth( double value ) const;
+
+    //! Writes configuration to a new DOM element
+    QDomElement writeXml( QDomDocument &doc, const QgsReadWriteContext &context ) const;
+    //! Reads configuration from the given DOM element
+    void readXml( const QDomElement &elem, const QgsReadWriteContext &context );
+
+  private:
+    bool mIsWidthVarying = false;
+
+    double mFixedWidth = DEFAULT_LINE_WIDTH;
+
+    double mMinimumValue = 0;
+    double mMaximumValue = 10;
+    double mMinimumWidth = DEFAULT_LINE_WIDTH;
+    double mMaximumWidth = 3;
+    bool mIgnoreOutOfRange = false;
+
+    mutable double mLinearCoef = 1;
+    mutable bool mNeedUpdateFormula = true;
+    void updateLinearFormula() const;
+};
+
+#ifndef SIP_RUN
+
+/**
+ * \ingroup core
+ *
+ * Class plotting mesh dataset with stroke with and color varying dependending on dataset value.
+ *
+ * \since QGIS 3.14
+ */
+class CORE_EXPORT QgsMeshPlotter
```

Comment:
Do we need to export it outside qgis_core? If used internally in mesh renderer I would keep it private and do not export it